xmtp_mls/src/groups/mls_sync.rs

```diff
@@ -1084,6 +1121,19 @@ where
             .db()
             .find_group_intent_by_payload_hash(&sha256(envelope.data.as_slice()))
             .map_err(GroupMessageProcessingError::Storage)?;
+
+        let group_cursor = provider
+            .db()
+            .get_last_cursor_for_id(&self.group_id, EntityKind::Group)?;
+        if group_cursor >= cursor as i64 {
+            // early return if the message is already procesed
+            // _NOTE_: Not early returning and re-processing a message that
+            // has already been processed, has the potential to result in forks.
+            return MessageIdentifierBuilder::from(envelope)
+                .previously_processed(true)
```

Comment:
is this field new/effect anything? if so, we should set it in the other places where we flag messages as previously processed